.repo/sample-list-generator/src/libs/api-detector.ts

```diff
@@ -0,0 +1,201 @@
+import {
+  ApiItem,
+  ApiItemWithType,
+  ApiTypeResult,
+  ExtensionApiMap
+} from '../types';
+import * as babel from '@babel/core';
+import { isIdentifier } from '@babel/types';
+import fs from 'fs/promises';
+import { getAllFiles } from '../utils/filesystem';
+import { loadExtensionApis } from './api-loader';
+
+let EXTENSION_API_MAP: ExtensionApiMap = loadExtensionApis();
+
+/**
+ * Gets the type of an api call.
+ * @param namespace - The namespace of the api call.
+ * @param propertyName - The property name of the api call.
+ * @returns The type of the api call.
+ * @example
+ * getApiType('tabs', 'query')
+ * // returns 'method'
+ */
+export const getApiType = (
+  namespace: string,
+  propertyName: string
+): ApiTypeResult => {
+  namespace = namespace.replace(/_/g, '.');
+
+  if (EXTENSION_API_MAP[namespace]) {
+    const apiTypes = EXTENSION_API_MAP[namespace];
+
+    if (apiTypes.methods.includes(propertyName)) {
+      return 'method';
+    }
+    if (apiTypes.events.includes(propertyName)) {
+      return 'event';
+    }
+    if (apiTypes.properties.includes(propertyName)) {
+      return 'property';
+    }
+    if (apiTypes.types.includes(propertyName)) {
+      return 'type';
+    }
+  }
+  return 'unknown';
+};
+
+/**
+ * Gets all the api calls in a sample.
+ * @param sampleFolderPath - The path to the sample folder.
+ * @returns A promise that resolves to an array of apis the sample uses.
+ */
+export const getApiListForSample = async (
+  sampleFolderPath: string
+): Promise<ApiItemWithType[]> => {
+  // get all js files in the folder
+  const jsFiles = (await getAllFiles(sampleFolderPath)).filter((file) =>
```

Comment:
You could move this into a method in the filesystem util.

Reply:
Done. https://github.com/GoogleChrome/chrome-extensions-samples/pull/963/commits/709b196051f32d94da42fc23ff21c7a96a8eccac